lib/engine/tile.rb

```diff
@@ -374,7 +374,9 @@ def add_reservation!(entity, city, slot = nil, reserve_city = true)
     def token_blocked_by_reservation?(corporation)
       return false if @reservations.empty?
 
-      if @reservation_blocks == :always || (@reservation_blocks == :yellow_only && @color == :yellow)
+      if @reservation_blocks == :always ||
+        (@reservation_blocks == :yellow_only && @color == :yellow) ||
+        (@reservation_blocks == :oo_only && @cities.size > 1)
```

Comment:
This isn't correct behavior. Why was this approved and merged?

Reply:
lack of familiarity probably. feel free to revert

Reply:
why is it wrong?

Reply:
@bentziaxl, the logic happens to work for 1880, but makes bad assumptions from a general reusability perspective . I provide a brief rationale here: #8879. Ping me on slack if you want to talk about it more.